src/openrct2/scenario/Scenario.cpp

```diff
@@ -473,15 +473,16 @@ static int32_t scenario_create_ducks()
     for (int32_t i = 0; i < duckCount; i++)
     {
         int32_t r = scenario_rand();
-        innerPos.x = (r >> 16) & 0x7F;
-        innerPos.y = (r & 0xFFFF) & 0x7F;
+        innerPos.x = (r >> 16) & (SquareRadiusSize - 1);
+        innerPos.y = (r & 0xFFFF) & (SquareRadiusSize - 1);
```

Comment:
Since squareRadiusSize is 96, it looks a bit strange to me to use & instead of %.

Reply:
The reason has been due to r being signed, I changed the type to unsigned.